text/src/autogluon/text/automm/utils.py

```diff
@@ -503,6 +508,72 @@ def create_model(
     else:
         raise ValueError(f"No available models for {names}")
 
+def save_pretrained_model(
+    model: ModuleList,
+    config: DictConfig,
+    path: str
+) -> DictConfig:
+    '''
+    Saving the pretrained weights for offline deployment. Used by turing "standalone=True" in predictor. 
+
+    Parameters
+    ----------
+    model
+        A modulelist containing all models.
+    config
+        A DictConfig object. The model config should be accessible by "config.model".
+    path
+        The saving path to the pretained weights i.e. config.json for "clip" and "hf_text", 'pkl' for "timm_image"
+    '''
+    for idx, model_name in enumerate(config.model.names):
+        if model_name == "clip" or "hf_text" in model_name:
+            model[idx].model.save_pretrained(os.path.join(path,model_name))
+            model_config = getattr(config.model, model_name)
+            model_config.checkpoint_name = os.path.join('local://',model_name)
+        if model_name == "timm_image":
+            model_config = getattr(config.model, model_name)
+            if not os.path.exists(os.path.join(path,model_name)):
+                os.makedirs(os.path.join(path,model_name))
+            torch.save(
+                model[idx].model.state_dict(),
+                os.path.join(path,model_name,model_config.checkpoint_name+'.pkl')
+            )
+            model_config.checkpoint_name = os.path.join('local://',model_name)
+    return config
+
+
+def load_pretrained_configs(
```

Comment:
Overall, this function mainly converts local path to global path. Adding `pretrained` in `config` also looks kind of confusing without deep understanding. I think we can avoid the path conversion by adding a new `local_path`  key under the hf_text and clip configs. Basically, we can do `checkpoint_name=os.path.join(local_path, checkpoint_name)` before passing `checkpoint_name` to HFAutoModelForTextPrediction and CLIPForImageText.

Reply:
If `local_path ` is None, it means downloading from Huggingface. So, no need additional flag `pretrained` to indicate that. Adding `local://` is also unnecessary f using `local_path`.

Reply:
Thanks for the great comments. I'll try into incorporate them.